benchmarks/common.py

```diff
@@ -1,18 +1,28 @@
 import os
 import timeit
+"""
+if environment variable $MULTICORE is set, benchmarks will be run for
+n_jobs vals = (1, $MULTICORE). Otherwise for n_jobs vals = (1, -1)
+MULTICORE can be a list of values as well.
+Example: export MULTICORE=2,3,4,5
+"""
 
+N_JOBS_VALS = [1, ]
 
-# if environment variable $MULTICORE is set, benchmarks will be run for
-# n_jobs vals = (1, $MULTICORE). Otherwise for n_jobs vals = (1, -1)
-N_JOBS_VALS = [1, int(os.getenv('MULTICORE', -1))]
+try:
+    multicore = os.getenv('MULTICORE')
+    results = map(int, multicore.split(','), -1)
+    N_JOBS_VALS.extend(results)
+except Exception:
+    pass
 
 
-class Benchmark():
+class Benchmark:
     timer = timeit.default_timer  # wall time
     timeout = 500
     processes = 1
     number = 1
     repeat = 1
```

Comment:
bad defaults for a benchmark. it means measure it once and use immediately. I don't remember which is which but I'd like to repeat it by default at least 3 times to see how it changes from run to run. there is also a good time-target option which tries to adjust the number of repeats to run more that a specified time.

Reply:
These are the defaults set up by Jeremy. I guess we can change them for our internal use

Reply:
please check `goal_time` and increase the number of outermost repeats